### DISCUSSION TITLE
Apex Error: ActivityHistories have an implementation restriction for non-admins

### DISCUSSION THREAD

**Discussion by korin-ly** [deejay-hub/timeline-lwc#192] [Q&A] [answered]
Hello!

Thank you for this wonderful tool, first of all. I'm running into an Apex error for non-admins (snapshot below). We unfortunately can't give them modify all access. We would like to display all activities related to an account, so we are using ActivityHistory. Is there a way around the implementation restriction? (It works as expected for admin users)

The error: "There is an implementation restriction on ActivityHistories. When you query this relationship, security evaluation is implemented for users who don't have administrator permissions, and you must use a specific sort order: ActivityDate DESC, LastModifiedDate DESC"

<img width="457" alt="ActivityHistoriesApexError" src="https://user-images.githubusercontent.com/106759136/171660534-3b4a8053-8d37-43a0-9814-d91b31d8b181.png">

Thank you!

**Comment by deejay-hub** [comment]
Hi @korin-ly . Just to confirm. Are you using the AppExchange package? if so what version have you got installed? Setup-> Apps -> Packaging -> Installed Packages.

**Reply by korin-ly** [reply]
v 1.10

<img width="737" alt="image" src="https://user-images.githubusercontent.com/106759136/171770430-e0baa1dd-5dd0-4af5-a72e-cac2535bdc88.png">

**Comment by deejay-hub** [comment]
Okay we'll take a look. Do you get the same error for OpenActivities? i.e. if you inactivate the metadata for ActivityHistory do you get the same error?

**Reply by korin-ly** [reply]
Hi! I removed ActivityHistory and the error doesn't occur. Thank you so much for looking into it!

**Reply by deejay-hub** [reply]
Thank you for taking the time to raise the issue.

**Comment by deejay-hub** [comment]
Bad news. Good news.

Bad news - it's a bug. Using ActivityHistories for non-admin users won't work. You'll have to use Tasks/Events instead for now (with the restriction on Accounts that it won't show all roll up activities from related records).

Good news - we can fix it in the next release. We plan to package the Luyten release v1.11 in the next 4 weeks (current date 27th June) and this fix can make the cut. So watch out for the release on AppExchange shortly after.
https://github.com/deejay-hub/timeline-lwc/issues/194

**Comment by korin-ly** [comment]
I think that is great news! Thank you so much! I'll keep an eye out.

**Comment by deejay-hub** [comment, marked as answer]
Fixed in v1.11. Available on the AppExchange.
